src/main.rs

```diff
@@ -0,0 +1,89 @@
+use std::path::{Path, PathBuf};
+
+use clap::{command, Parser, Subcommand};
+use wallpaper::{DesktopWallpaper, DesktopWallpaperPosition, Monitor};
+
+/// Manage the desktop wallpaper on Windows
+#[derive(Debug, Parser)]
+#[command(author, version, about, long_about = None)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Debug, Subcommand)]
+enum Commands {
+    /// Gets current wallpaper
+    Get {
+        /// Index of monitor starting from 0
+        #[arg(short, long, default_value_t = 0)]
+        monitor: usize,
+    },
+
+    /// Sets the current wallpaper
+    #[command(arg_required_else_help = true)]
+    Set {
+        /// Path to wallpaper
+        path: PathBuf,
+        /// Specifies how the desktop wallpaper should be displayed.
+        #[arg(
+            short,
+            long,
+            default_value_t = DesktopWallpaperPosition::Span,
+            default_missing_value = "always",
+            value_enum
+        )]
+        scale: DesktopWallpaperPosition,
+        /// Index of monitor starting from 0
+        #[arg(short, long, default_value_t = 0)]
+        monitor: usize,
+    },
+}
+
+fn if_chosen_monitor_within_range(choice: usize, monitors: &[Monitor], func: impl Fn()) {
+    if choice < monitors.len() {
+        func();
+    } else {
+        eprintln!(
+            "The available monitors are from 0 to {} but {choice} was given",
+            monitors.len() - 1,
+        );
+    }
+}
```

Comment:
```suggestion
fn if_chosen_monitor_within_range(choice: usize, monitors: &[Monitor], func: impl FnOnce(&Monitor)) {
    match monitors.get(choice) {
        Some(m) => func(m),
        None => eprintln!(
            "The available monitors are from 0 to {} but {choice} was given",
            monitors.len() - 1,
        ),
    }
}
```

avoids extra bounds checking when you access the index